syft/format/decoders_collection.go

```diff
@@ -45,10 +52,16 @@ func (c *DecoderCollection) Decode(reader io.Reader) (*sbom.SBOM, sbom.FormatID,
 }
 
 // Identify takes a set of bytes and attempts to identify the format of the SBOM relative to the decoders in the collection.
-func (c *DecoderCollection) Identify(reader io.Reader) (sbom.FormatID, string) {
-	if reader == nil {
+func (c *DecoderCollection) Identify(r io.Reader) (sbom.FormatID, string) {
+	if r == nil {
 		return "", ""
 	}
+
+	reader, err := stream.SeekableReader(r)
+	if err != nil {
```

Comment:
this should probably get logged with `log.Debugf("unable to create seekable reader: %v", err)` or similar

Reply:
[Addressed](https://github.com/anchore/syft/compare/445629088bc76ff4a4fd5f83e582dcde26c9b34f..c0380cecf43580fbc2de65959d826219ad5c69a3)